enterprise/admin/__init__.py

```diff
@@ -135,12 +167,89 @@ class EnterpriseCustomerUserAdmin(admin.ModelAdmin):
     class Meta(object):
         model = EnterpriseCustomerUser
 
-    fields = ('user_id', 'enterprise_customer')
+    fields = (
+        'user_id',
+        'enterprise_customer',
+        'user_email',
+        'username',
+        'created',
+        'enrolled_courses',
+    )
+
+    # Only include fields that are not database-backed; DB-backed fields
+    # are dynamically set as read only.
+    readonly_fields = (
+        'user_email',
+        'username',
+        'created',
+        'enrolled_courses',
+    )
+
+    inlines = (
+        HistoricalUserDataSharingConsentAuditInlineAdmin,
+    )
+
+    def username(self, ecu):
+        """
+        Return the username for the attached user.
+        """
+        return ecu.user.username
+
+    def enrolled_courses(self, ecu):
+        """
+        Return a string represented the courses enrolled in
+        """
+        courses_string = mark_safe(self.get_enrolled_course_string(ecu))
+        return courses_string or 'None'
 
     def get_readonly_fields(self, request, obj=None):
         """
         Make all fields readonly when editing existing model.
         """
+        readonly_fields = super(EnterpriseCustomerUserAdmin, self).get_readonly_fields(request, obj=obj)
         if obj:  # editing an existing object
-            return get_all_field_names(self.model)
-        return tuple()
+            return readonly_fields + tuple(get_all_field_names(self.model))
+        return readonly_fields
+
+    def get_enrolled_course_string(self, ecu):
+        """
+        Get an HTML string representing the courses the user is enrolled in.
+        """
+        enrolled_courses = get_enrolled_courses(user=self.username(ecu))
+        course_details = []
+        for course in enrolled_courses:
+            course_id = course['course_details']['course_id']
```

Comment:
The values here are pretty much fundamental to the data we're receiving; while we could suppress potential errors by doing a `.get()` call, this is one case where I'd never _expect_ to get a `KeyError`, so if there is one, as a platform administrator, I'd like to know right away, and loudly. One example of such a case would be if the API client returned a value when it should have raised its own exception instead.

I'm certainly open to making whatever change you request, though.